src/api/eventTypes.js

```diff
@@ -173,3 +175,236 @@ export type RestartEvent = $ReadOnly<{|
   zulip_version?: string,
   zulip_feature_level?: number,
 |}>;
+
+/* eslint-disable flowtype/generic-spacing */
+
+/**
+ * Data updates we might get in `update` or `update_dict` realm events.
+ */
+// If there's a property that exists in InitialDataRealm, but we don't use
+// it and the event doc doesn't mention it, write the InitialDataRealm
+// property in a code comment in the place where it would go here.
+type RealmDataForUpdate = $ReadOnly<{
+  //
+  // Keep alphabetical by the InitialDataRealm property. So by
+  // realm_allow_edit_history, not by allow_edit_history, etc.
+  //
+
+  development_environment: $PropertyType<InitialDataRealm, 'development_environment'>,
+
+  // TODO(server-5.0): See note in InitialDataRealm.
+  // event_queue_longpoll_timeout_seconds
+
+  jitsi_server_url: $PropertyType<InitialDataRealm, 'jitsi_server_url'>,
+  max_avatar_file_size_mib: $PropertyType<InitialDataRealm, 'max_avatar_file_size_mib'>,
+  max_file_upload_size_mib: $PropertyType<InitialDataRealm, 'max_file_upload_size_mib'>,
+
+  // TODO: Check if the doc really means this, instead of max_icon_file_size
+  // and max_icon_file_size_mib?
+  icon_file_size: number,
+
+  // TODO(server-5.0): See notes in InitialDataRealm.
+  // max_logo_file_size
+  // max_logo_file_size_mib
+
+  password_min_length: $PropertyType<InitialDataRealm, 'password_min_length'>,
+  password_min_guesses: $PropertyType<InitialDataRealm, 'password_min_guesses'>,
+
+  // TODO(server-5.0): See notes in InitialDataRealm.
+  // realm_add_custom_emoji_policy
+  // realm_add_emoji_by_admins_only
+  // realm_allow_community_topic_editing
+
+  allow_edit_history: $PropertyType<InitialDataRealm, 'realm_allow_edit_history'>,
+
+  // TODO(server-5.0): See note in InitialDataRealm.
+  // realm_allow_message_deleting
+
+  allow_message_editing: $PropertyType<InitialDataRealm, 'realm_allow_message_editing'>,
+  authentication_methods: $PropertyType<InitialDataRealm, 'realm_authentication_methods'>,
+
+  // realm_available_video_chat_providers doesn't have an event because it
+  //   can only change when the server is restarted anyway; see
+  //   https://chat.zulip.org/#narrow/stream/378-api-design/topic/.60value.60.20in.20realm.20op.3A.20update/near/1276269
+
+  avatar_changes_disabled: $PropertyType<InitialDataRealm, 'realm_avatar_changes_disabled'>,
+  bot_creation_policy: $PropertyType<InitialDataRealm, 'realm_bot_creation_policy'>,
+  bot_domain: $PropertyType<InitialDataRealm, 'realm_bot_domain'>,
+
+  // TODO(server-3.0): See note in InitialDataRealm.
```

Comment:
For these comments that just refer to similar comments in InitialDataRealm, let's omit this copy. That'll help it be more uniform to read through, and let the more specific comments like on `realm_available_video_chat_providers` stand out better.